plugins/woocommerce-admin/client/products/fields/attribute-term-input-field/create-attribute-term-modal.tsx

```diff
@@ -0,0 +1,162 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Button,
+	Modal,
+	TextareaControl,
+	TextControl,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { cleanForSlug } from '@wordpress/url';
+import { Form, FormContext, FormErrors } from '@woocommerce/components';
+import { recordEvent } from '@woocommerce/tracks';
+import {
+	EXPERIMENTAL_PRODUCT_ATTRIBUTE_TERMS_STORE_NAME,
+	ProductAttributeTerm,
+	QueryProductAttribute,
+} from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import './create-attribute-term-modal.scss';
+
+type CreateAttributeTermModalProps = {
+	initialAttributeTermName: string;
+	attributeId: number;
+	onCancel: () => void;
```

Comment:
I'm wondering if there's a possible use case in which these handlers should be optional?

Reply:
I don't think so, but I made them optional just incase, I defaulted them to empty functions in this commit: https://github.com/woocommerce/woocommerce/pull/35131/commits/2f39f27c9fc5e41179fc13f8e92f5c3d4a1c6b5d